packages/editor/src/components/rich-text/format-edit.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * WordPress dependencies
+ */
+import { Component, Fragment } from '@wordpress/element';
+import { getActiveFormat, getFormatTypes } from '@wordpress/rich-text';
+import { Fill, KeyboardShortcuts, ToolbarButton } from '@wordpress/components';
+import { rawShortcut } from '@wordpress/keycodes';
+
+/**
+ * Internal dependencies
+ */
+import InserterListItem from '../inserter-list-item';
+import { normalizeTerm } from '../inserter/menu';
+
+function isResult( { title, keywords = [] }, filterValue ) {
+	const normalizedSearchTerm = normalizeTerm( filterValue );
+	const matchSearch = ( string ) => normalizeTerm( string ).indexOf( normalizedSearchTerm ) !== -1;
+	return matchSearch( title ) || keywords.some( matchSearch );
+}
+
+function FillToolbarButton( { name, ...props } ) {
+	return (
+		<Fill name={ `RichText.ToolbarControls.${ name }` }>
+			<ToolbarButton { ...props } />
+		</Fill>
+	);
+}
+
+function FillInserterListItem( props ) {
+	return (
+		<Fill name="Inserter.InlineElements">
+			{ ( { filterValue } ) => {
+				if ( filterValue && ! isResult( props, filterValue ) ) {
+					return null;
+				}
+
+				return <InserterListItem { ...props } />;
+			} }
+		</Fill>
+	);
+}
+
+class Shortcut extends Component {
+	constructor() {
+		super( ...arguments );
+
+		this.onUse = this.onUse.bind( this );
+	}
+
+	onUse() {
+		this.props.onUse();
+		return false;
+	}
+
+	render() {
+		const { character, type } = this.props;
+
+		return (
+			<KeyboardShortcuts
+				bindGlobal
+				shortcuts={ {
+					[ rawShortcut[ type ]( character ) ]: this.onUse,
+				} }
+			/>
+		);
+	}
+}
+
+const FormatEdit = ( { onChange, value } ) => {
+	return (
+		<Fragment>
+			{ getFormatTypes().map( ( { name, edit: Edit, keywords } ) => {
```

Comment:
I'm coming to the conclusion that we should be using `withSelect` to ensure this component re-renders every time someone register/unregisters a format type.

Reply:
I think this is why when I unregister format I need to do some UI interactions before it gets removed.

Reply:
Not a blocker! :)

Reply:
Sounds good!